packages/filter-workspace-packages/src/getChangedPackages.ts

```diff
@@ -1,15 +1,15 @@
 import PnpmError from '@pnpm/error'
+import * as micromatch from 'micromatch'
 import path = require('path')
 import execa = require('execa')
 import findUp = require('find-up')
 import isSubdir = require('is-subdir')
 
-export default async function changedSince (packageDirs: string[], commit: string, opts: { workspaceDir: string }): Promise<string[]> {
+export default async function changedSince (packageDirs: string[], commit: string, opts: { workspaceDir: string, testPattern?: string }): Promise<[string[], string[]]> {
   const repoRoot = path.resolve(await findUp('.git', { cwd: opts.workspaceDir, type: 'directory' }) ?? opts.workspaceDir, '..')
-  let changedDirs = Array.from(
-    await getChangedDirsSinceCommit(commit, opts.workspaceDir)
-  ).map(changedDir => path.join(repoRoot, changedDir))
-  const changedPkgs = []
+  const [dirsWithChanges, dirsMatchingFilter] = await getChangedDirsSinceCommit(commit, opts.workspaceDir, opts.testPattern)
```

Comment:
A better name would probably be `dirsWithTestChanges` instead of `dirsMatchingFilter`

Reply:
done [here](https://github.com/pnpm/pnpm/pull/3035/commits/5678d4d2a7a1a661f81602c323d01684a5e307b5)